packages/sdk/src/utils/__test__/helpers.test.ts

```diff
@@ -30,16 +38,50 @@ describe('constructDepositDataEvmSubstrate', () => {
     const expectedDepositData = '0x1234';
 
     jest.spyOn(utils, 'parseUnits').mockReturnValueOnce(BigNumber.from('100000000000000000000'));
-    jest.spyOn(utils, 'isAddress').mockReturnValueOnce(true);
-    jest.spyOn(utils, 'arrayify').mockReturnValueOnce(new Uint8Array([18, 52, 86]));
+    jest
+      .spyOn(helpers, 'getRecipientAddressInBytes')
+      .mockReturnValueOnce(new Uint8Array([18, 52, 86]));
     jest.spyOn(utils, 'hexlify').mockReturnValueOnce(expectedDepositData);
 
-    const depositData = constructDepositDataEvmSubstrate(tokenAmount, recipientAddress, decimals);
+    const depositData = createERCDepositData(tokenAmount, recipientAddress, decimals);
 
     expect(utils.parseUnits).toHaveBeenCalledWith(tokenAmount, decimals);
-    expect(utils.isAddress).toHaveBeenCalledWith(recipientAddress);
-    expect(utils.arrayify).toHaveBeenCalledWith(recipientAddress);
+    expect(helpers.getRecipientAddressInBytes).toHaveBeenCalledWith(recipientAddress);
     expect(utils.hexlify).toHaveBeenCalledWith(expectedBytesArr);
     expect(depositData).toEqual(expectedDepositData);
   });
 });
+
+describe('getRecipientAddressInBytes', () => {
+  it('should convert an EVM address to a Uint8Array of bytes', () => {
+    const evmAddress = '0x742d35Cc6634C0532925a3b844Bc454e4438f44e';
+    expect(utils.isAddress(evmAddress)).toBeTruthy();
+
+    const result = getRecipientAddressInBytes(evmAddress);
+    const expectedResult = utils.arrayify(evmAddress);
+
+    expect(result).toEqual(expectedResult);
+  });
+
+  it('should convert a Substrate multilocation to a Uint8Array of bytes', () => {
```

Comment:
Same thing as above, and I assume `0x06a220edf5f82b84fc5f9270f8a30a17636bf29c05a5c16279405ca20918aa39` is substrate equivalent of evm `0x742d35Cc6634C0532925a3b844Bc454e4438f44e` and I don't see the logic that does that coversion.

Reply:
I do not convert addresses here. I convert json-like object of multilocation to string, and then to bytes. Then these data will be hexified with other params and passed as depositData to the contract.

Reply:
added `expect(result).toBeInstanceOf(Uint8Array);`. Same as above

Reply:
It is public key of substrate address

Reply:
The address `5CDQJk6kxvBcjauhrogUc9B8vhbdXhRscp1tGEUmniryF1Vt` equals to `0x06a220edf5f82b84fc5f9270f8a30a17636bf29c05a5c16279405ca20918aa39`
https://stackoverflow.com/questions/60983883/how-to-encode-the-hex-string-representation-of-an-account-id-in-substrate-using

Reply:
@freddyli7 please correct me if use wrong terminology here